File: testprojects/java/accuracy/domain/indirect/violatingfrom/AccessObjectReferenceIndirect_AsParameter_POI.java
package domain.indirect.violatingfrom;

import domain.indirect.intermediate.BackgroundService;
import domain.indirect.intermediate.ServiceTwo;

public class AccessObjectReferenceIndirect_AsParameter_POI {
	
	private BackgroundService bckg = new BackgroundService();
	private ServiceTwo st = new ServiceTwo();

	public AccessObjectReferenceIndirect_AsParameter_POI(){
		String s = bckg.getPOINameByArgument(st.getServiceOne().poi);
		System.out.println(s);
	}
}

File: src/test/resources/csharp/benchmark/Domain/Indirect/AllowedFrom/AccessObjectReferenceIndirect_AsParameter_POI.cs
namespace Domain.Indirect.AllowedFrom
{
	public class AccessObjectReferenceIndirect_AsParameter_POI
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg = new Domain.Indirect.Intermediate.BackgroundService();

        private Domain.Indirect.Intermediate.ServiceTwo st = new Domain.Indirect.Intermediate.ServiceTwo();

		public AccessObjectReferenceIndirect_AsParameter_POI()
		{
			string s = bckg.GetPOINameByArgument(st.GetServiceOne().poi);
		}
	}
}
